python/ray/data/tests/test_dataset.py

```diff
@@ -3260,6 +3260,30 @@ def flaky_mapper(x):
         ).take()
 
 
+def test_dataset_name(ray_start_regular_shared, tmp_path):
+    ds = ray.data.range(10, _dataset_name="MyDataset")
+    assert ds._get_name() == "MyDataset"
+    ds = ds.random_shuffle()
+    assert ds._get_name() == "MyDataset|random_shuffle"
```

Comment:
Use ":" instead of "|"? We already use ":" in task names.

Reply:
Hmm I'm going back and forth on this one, since using `"|"` is nice for connoting chained operations... These will be prefixed with `ray::` when displayed, so it might be nice to use a different separator. A few options:
```python
ds = ray.data.range(10, _dataset_name="MyDataset").map().filter().random_shuffle()
```
### Option 1: `|`
```
ray::MyDataset|map|filter|random_shuffle
```
### Option 2: `:`
```
ray::MyDataset:map:filter:random_shuffle
```
### Option 3: `-`
```
ray::MyDataset-map-filter-random_shuffle
```
I find options 1 and 3 to be the most readable. FWIW, Dask uses option 3 for their task keys containing chained operations: https://docs.dask.org/en/stable/10-minutes-to-dask.html#visualize-the-task-graph